ILL/IDF/d7_generateIDF.py

```diff
@@ -0,0 +1,133 @@
+'''
+Created on 04/05/2020
+
+
+
+'''
+
+
+import os
+path = os.path.abspath("")
+import sys
+sys.path.insert(0, path)
+
+from helper import MantidGeom
+from lxml import etree as le
+import math
+
+# using metre as unit
+instrumentName = 'D7'
+validFrom = "1900-01-31 23:59:59"
+validTo="2100-01-31 23:59:59"
+
+# Monochromator position
+source = -3.6
+
+# Fermi chopper
+zChopper = -0.48
+
+# 2 Monitors
+zMon1 = -1.3
+zMon2 = 1.5
+
+# distance between the sample and detectors:
+rSampleDetectors = 1.5177 # distance from the sample of even detectors, indices start from 0 (even)
+minTwoTheta = 0
+maxTwoTheta = 136
+
+# definition of a detector bank
+numberBanks = 3 # banks 2, 3 and 4; bank 1 is a monitor
+numberDetectors = 44
+
+# definition of a pixel
+tubeRadius = 0.01252
+tubeHeight = 0.25
+
+# definitions of intersection positions for detectors in the same casette
+externalIntersectonDistance = 0.011
+intersectionAngle = 180.0 * externalIntersectonDistance / (2.0 * math.pi * (rSampleDetectors - tubeRadius))   
+detectorOffset = 1.114 # degrees
+
+
+comment = """ This is the instrument definition file of the D7 diffuse scattering spectrometer at the ILL.
+       Generated file, PLEASE DO NOT EDIT THIS FILE!
+       This file was automatically generated by mantidgeometry/ILL/IDF/d7_generateIDF.py
+       The generating script should be run as:
+       cd mantidgeometry/ILL/IDF; python d7_generateIDF.py
```

Comment:
I think that's not the way to run. You have to be in the root to be able to import the helper, so:
`python3 ./ILL/IDF/d7_generateIDF.py`?